.github/workflows/build.yml

```diff
@@ -7,53 +7,74 @@ on:
     branches: [ 2.x, develop, master ]
 
 jobs:
-  build_arm64-binary: 
-     runs-on: ubuntu-latest
-     if: ${{ github.event_name == 'push' && (github.ref == 'refs/heads/develop' || github.ref == 'refs/heads/2.x') }}
-     strategy:
+  # job 1: Compile based on java11, and checkstyle.
+  compile:
+    name: "compile"
```

Comment:
I think the hierarchy is clearer and simpler without splitting(`compile` and  `build` )

Reply:
done, remove the `compile` job.